helsingborg-stad/helsingborg-io-sls-api
# Feat/set initial case answers

## Explain the changes you’ve made
Added functionality to pre populate case answers with previous case answers and user information. 

## Explain your solution
Created a method that takes four arguments (initial forms, user object, previous case & form templates) and populates the initial forms with previous case data and user data. 
Which fields that will be pre populated can be defined from the formbuilder.

**Note** that the feature to load previous data is only activated to some of the fields(text, number, checkbox, repeaterField and editableList) in the formbuilder and will not work for rest of the fields. I'm not sure why we don't have it activated for all fields, could be a future task to fix. 

This new functionality is used by createCase + createVivaCase lambdas during the creation of new cases.

I have chosen to get previous case data from the latest updated case that also has status set to "closed". So this will not get previous answers from "ongoing" cases for example.

## How to test the changes?

First make sure you have a user, forms and a case(with status: closed) in your DB.
The form needs to have fields with "load previous" functionality activated. 

Deploy case and viva services. 
Make a POST request to create case endpoint.
Example body json:
`
{
  "statusType": "notStarted",
  "currentFormId": "SOME-FORM-ID",
  "provider": "VIVA",
  "details": {},
  "forms": {
    "SOME-FORM-ID {
      "answers": [],
      "currentPosition": {
        "index": 1,
        "level": 0,
        "currentMainStep": 0,
        "currentMainStepIndex": 0
      }
    }
  }
}
` 

If everything works the answers object should be populated with the data you have defined in form builder.  

## Anything else
I have only tested this with the createCase endpoint, because I dont know how to trigger event that creates a new VIVA case. So it would be great if someone could test that too.
